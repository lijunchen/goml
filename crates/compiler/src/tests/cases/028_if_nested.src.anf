fn classify(x/0: int) -> string {
  let t6 = int_less(x/0, 0) in
  if t6 {
    "negative"
  } else {
    let t7 = int_less(0, x/0) in
    if t7 {
      "positive"
    } else {
      "zero"
    }
  }
}

fn triangle_type(a/1: int, b/2: int, c/3: int) -> string {
  let t8 = int_less(a/1, b/2) in
  if t8 {
    let t9 = int_less(b/2, c/3) in
    if t9 {
      "ascending"
    } else {
      "peak"
    }
  } else {
    let t10 = int_less(a/1, c/3) in
    if t10 {
      "valley"
    } else {
      "flat"
    }
  }
}

fn main() -> unit {
  let t11 = int_neg(42) in
  let first/4 = classify(t11) in
  let second/5 = classify(0) in
  let third/6 = classify(17) in
  let shape1/7 = triangle_type(1, 2, 3) in
  let shape2/8 = triangle_type(3, 2, 1) in
  let shape3/9 = triangle_type(2, 3, 2) in
  let mtmp0 = string_println(first/4) in
  let mtmp1 = string_println(second/5) in
  let mtmp2 = string_println(third/6) in
  let mtmp3 = string_println(shape1/7) in
  let mtmp4 = string_println(shape2/8) in
  let mtmp5 = string_println(shape3/9) in
  ()
}