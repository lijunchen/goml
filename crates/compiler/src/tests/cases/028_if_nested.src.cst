FILE@0..845
  FN@0..160
    FnKeyword@0..2 "fn"
    Whitespace@2..3 " "
    Lident@3..11 "classify"
    PARAM_LIST@11..20
      LParen@11..12 "("
      PARAM@12..18
        Lident@12..13 "x"
        Colon@13..14 ":"
        Whitespace@14..15 " "
        TYPE_INT@15..18
          IntKeyword@15..18 "int"
      RParen@18..19 ")"
      Whitespace@19..20 " "
    Arrow@20..22 "->"
    Whitespace@22..23 " "
    TYPE_STRING@23..30
      StringKeyword@23..29 "string"
      Whitespace@29..30 " "
    BLOCK@30..160
      LBrace@30..31 "{"
      Whitespace@31..34 "\n  "
      EXPR_IF@34..157
        IfKeyword@34..36 "if"
        Whitespace@36..37 " "
        EXPR_IF_COND@37..52
          EXPR_CALL@37..52
            EXPR_LIDENT@37..45
              Lident@37..45 "int_less"
            ARG_LIST@45..52
              LParen@45..46 "("
              ARG@46..49
                EXPR_LIDENT@46..47
                  Lident@46..47 "x"
                Comma@47..48 ","
                Whitespace@48..49 " "
              ARG@49..50
                EXPR_INT@49..50
                  Int@49..50 "0"
              RParen@50..51 ")"
              Whitespace@51..52 " "
        EXPR_IF_THEN@52..73
          BLOCK@52..73
            LBrace@52..53 "{"
            Whitespace@53..58 "\n    "
            EXPR_STR@58..71
              Str@58..68 "\"negative\""
              Whitespace@68..71 "\n  "
            RBrace@71..72 "}"
            Whitespace@72..73 " "
        ElseKeyword@73..77 "else"
        Whitespace@77..78 " "
        EXPR_IF_ELSE@78..157
          BLOCK@78..157
            LBrace@78..79 "{"
            Whitespace@79..84 "\n    "
            EXPR_IF@84..155
              IfKeyword@84..86 "if"
              Whitespace@86..87 " "
              EXPR_IF_COND@87..102
                EXPR_CALL@87..102
                  EXPR_LIDENT@87..95
                    Lident@87..95 "int_less"
                  ARG_LIST@95..102
                    LParen@95..96 "("
                    ARG@96..99
                      EXPR_INT@96..97
                        Int@96..97 "0"
                      Comma@97..98 ","
                      Whitespace@98..99 " "
                    ARG@99..100
                      EXPR_LIDENT@99..100
                        Lident@99..100 "x"
                    RParen@100..101 ")"
                    Whitespace@101..102 " "
              EXPR_IF_THEN@102..127
                BLOCK@102..127
                  LBrace@102..103 "{"
                  Whitespace@103..110 "\n      "
                  EXPR_STR@110..125
                    Str@110..120 "\"positive\""
                    Whitespace@120..125 "\n    "
                  RBrace@125..126 "}"
                  Whitespace@126..127 " "
              ElseKeyword@127..131 "else"
              Whitespace@131..132 " "
              EXPR_IF_ELSE@132..155
                BLOCK@132..155
                  LBrace@132..133 "{"
                  Whitespace@133..140 "\n      "
                  EXPR_STR@140..151
                    Str@140..146 "\"zero\""
                    Whitespace@146..151 "\n    "
                  RBrace@151..152 "}"
                  Whitespace@152..155 "\n  "
            RBrace@155..156 "}"
            Whitespace@156..157 "\n"
      RBrace@157..158 "}"
      Whitespace@158..160 "\n\n"
  FN@160..398
    FnKeyword@160..162 "fn"
    Whitespace@162..163 " "
    Lident@163..176 "triangle_type"
    PARAM_LIST@176..201
      LParen@176..177 "("
      PARAM@177..185
        Lident@177..178 "a"
        Colon@178..179 ":"
        Whitespace@179..180 " "
        TYPE_INT@180..183
          IntKeyword@180..183 "int"
        Comma@183..184 ","
        Whitespace@184..185 " "
      PARAM@185..193
        Lident@185..186 "b"
        Colon@186..187 ":"
        Whitespace@187..188 " "
        TYPE_INT@188..191
          IntKeyword@188..191 "int"
        Comma@191..192 ","
        Whitespace@192..193 " "
      PARAM@193..199
        Lident@193..194 "c"
        Colon@194..195 ":"
        Whitespace@195..196 " "
        TYPE_INT@196..199
          IntKeyword@196..199 "int"
      RParen@199..200 ")"
      Whitespace@200..201 " "
    Arrow@201..203 "->"
    Whitespace@203..204 " "
    TYPE_STRING@204..211
      StringKeyword@204..210 "string"
      Whitespace@210..211 " "
    BLOCK@211..398
      LBrace@211..212 "{"
      Whitespace@212..215 "\n  "
      EXPR_IF@215..395
        IfKeyword@215..217 "if"
        Whitespace@217..218 " "
        EXPR_IF_COND@218..233
          EXPR_CALL@218..233
            EXPR_LIDENT@218..226
              Lident@218..226 "int_less"
            ARG_LIST@226..233
              LParen@226..227 "("
              ARG@227..230
                EXPR_LIDENT@227..228
                  Lident@227..228 "a"
                Comma@228..229 ","
                Whitespace@229..230 " "
              ARG@230..231
                EXPR_LIDENT@230..231
                  Lident@230..231 "b"
              RParen@231..232 ")"
              Whitespace@232..233 " "
        EXPR_IF_THEN@233..313
          BLOCK@233..313
            LBrace@233..234 "{"
            Whitespace@234..239 "\n    "
            EXPR_IF@239..311
              IfKeyword@239..241 "if"
              Whitespace@241..242 " "
              EXPR_IF_COND@242..257
                EXPR_CALL@242..257
                  EXPR_LIDENT@242..250
                    Lident@242..250 "int_less"
                  ARG_LIST@250..257
                    LParen@250..251 "("
                    ARG@251..254
                      EXPR_LIDENT@251..252
                        Lident@251..252 "b"
                      Comma@252..253 ","
                      Whitespace@253..254 " "
                    ARG@254..255
                      EXPR_LIDENT@254..255
                        Lident@254..255 "c"
                    RParen@255..256 ")"
                    Whitespace@256..257 " "
              EXPR_IF_THEN@257..283
                BLOCK@257..283
                  LBrace@257..258 "{"
                  Whitespace@258..265 "\n      "
                  EXPR_STR@265..281
                    Str@265..276 "\"ascending\""
                    Whitespace@276..281 "\n    "
                  RBrace@281..282 "}"
                  Whitespace@282..283 " "
              ElseKeyword@283..287 "else"
              Whitespace@287..288 " "
              EXPR_IF_ELSE@288..311
                BLOCK@288..311
                  LBrace@288..289 "{"
                  Whitespace@289..296 "\n      "
                  EXPR_STR@296..307
                    Str@296..302 "\"peak\""
                    Whitespace@302..307 "\n    "
                  RBrace@307..308 "}"
                  Whitespace@308..311 "\n  "
            RBrace@311..312 "}"
            Whitespace@312..313 " "
        ElseKeyword@313..317 "else"
        Whitespace@317..318 " "
        EXPR_IF_ELSE@318..395
          BLOCK@318..395
            LBrace@318..319 "{"
            Whitespace@319..324 "\n    "
            EXPR_IF@324..393
              IfKeyword@324..326 "if"
              Whitespace@326..327 " "
              EXPR_IF_COND@327..342
                EXPR_CALL@327..342
                  EXPR_LIDENT@327..335
                    Lident@327..335 "int_less"
                  ARG_LIST@335..342
                    LParen@335..336 "("
                    ARG@336..339
                      EXPR_LIDENT@336..337
                        Lident@336..337 "a"
                      Comma@337..338 ","
                      Whitespace@338..339 " "
                    ARG@339..340
                      EXPR_LIDENT@339..340
                        Lident@339..340 "c"
                    RParen@340..341 ")"
                    Whitespace@341..342 " "
              EXPR_IF_THEN@342..365
                BLOCK@342..365
                  LBrace@342..343 "{"
                  Whitespace@343..350 "\n      "
                  EXPR_STR@350..363
                    Str@350..358 "\"valley\""
                    Whitespace@358..363 "\n    "
                  RBrace@363..364 "}"
                  Whitespace@364..365 " "
              ElseKeyword@365..369 "else"
              Whitespace@369..370 " "
              EXPR_IF_ELSE@370..393
                BLOCK@370..393
                  LBrace@370..371 "{"
                  Whitespace@371..378 "\n      "
                  EXPR_STR@378..389
                    Str@378..384 "\"flat\""
                    Whitespace@384..389 "\n    "
                  RBrace@389..390 "}"
                  Whitespace@390..393 "\n  "
            RBrace@393..394 "}"
            Whitespace@394..395 "\n"
      RBrace@395..396 "}"
      Whitespace@396..398 "\n\n"
  FN@398..845
    FnKeyword@398..400 "fn"
    Whitespace@400..401 " "
    Lident@401..405 "main"
    PARAM_LIST@405..408
      LParen@405..406 "("
      RParen@406..407 ")"
      Whitespace@407..408 " "
    BLOCK@408..845
      LBrace@408..409 "{"
      Whitespace@409..412 "\n  "
      EXPR_LET@412..843
        LetKeyword@412..415 "let"
        Whitespace@415..416 " "
        PATTERN_VARIABLE@416..422
          Lident@416..421 "first"
          Whitespace@421..422 " "
        Eq@422..423 "="
        Whitespace@423..424 " "
        EXPR_LET_VALUE@424..438
          EXPR_CALL@424..438
            EXPR_LIDENT@424..432
              Lident@424..432 "classify"
            ARG_LIST@432..438
              LParen@432..433 "("
              ARG@433..436
                EXPR_PREFIX@433..436
                  Minus@433..434 "-"
                  EXPR_INT@434..436
                    Int@434..436 "42"
              RParen@436..437 ")"
              Whitespace@437..438 " "
        InKeyword@438..440 "in"
        Whitespace@440..443 "\n  "
        EXPR_LET_BODY@443..843
          EXPR_LET@443..843
            LetKeyword@443..446 "let"
            Whitespace@446..447 " "
            PATTERN_VARIABLE@447..454
              Lident@447..453 "second"
              Whitespace@453..454 " "
            Eq@454..455 "="
            Whitespace@455..456 " "
            EXPR_LET_VALUE@456..468
              EXPR_CALL@456..468
                EXPR_LIDENT@456..464
                  Lident@456..464 "classify"
                ARG_LIST@464..468
                  LParen@464..465 "("
                  ARG@465..466
                    EXPR_INT@465..466
                      Int@465..466 "0"
                  RParen@466..467 ")"
                  Whitespace@467..468 " "
            InKeyword@468..470 "in"
            Whitespace@470..473 "\n  "
            EXPR_LET_BODY@473..843
              EXPR_LET@473..843
                LetKeyword@473..476 "let"
                Whitespace@476..477 " "
                PATTERN_VARIABLE@477..483
                  Lident@477..482 "third"
                  Whitespace@482..483 " "
                Eq@483..484 "="
                Whitespace@484..485 " "
                EXPR_LET_VALUE@485..498
                  EXPR_CALL@485..498
                    EXPR_LIDENT@485..493
                      Lident@485..493 "classify"
                    ARG_LIST@493..498
                      LParen@493..494 "("
                      ARG@494..496
                        EXPR_INT@494..496
                          Int@494..496 "17"
                      RParen@496..497 ")"
                      Whitespace@497..498 " "
                InKeyword@498..500 "in"
                Whitespace@500..503 "\n  "
                EXPR_LET_BODY@503..843
                  EXPR_LET@503..843
                    LetKeyword@503..506 "let"
                    Whitespace@506..507 " "
                    PATTERN_VARIABLE@507..514
                      Lident@507..513 "shape1"
                      Whitespace@513..514 " "
                    Eq@514..515 "="
                    Whitespace@515..516 " "
                    EXPR_LET_VALUE@516..539
                      EXPR_CALL@516..539
                        EXPR_LIDENT@516..529
                          Lident@516..529 "triangle_type"
                        ARG_LIST@529..539
                          LParen@529..530 "("
                          ARG@530..533
                            EXPR_INT@530..531
                              Int@530..531 "1"
                            Comma@531..532 ","
                            Whitespace@532..533 " "
                          ARG@533..536
                            EXPR_INT@533..534
                              Int@533..534 "2"
                            Comma@534..535 ","
                            Whitespace@535..536 " "
                          ARG@536..537
                            EXPR_INT@536..537
                              Int@536..537 "3"
                          RParen@537..538 ")"
                          Whitespace@538..539 " "
                    InKeyword@539..541 "in"
                    Whitespace@541..544 "\n  "
                    EXPR_LET_BODY@544..843
                      EXPR_LET@544..843
                        LetKeyword@544..547 "let"
                        Whitespace@547..548 " "
                        PATTERN_VARIABLE@548..555
                          Lident@548..554 "shape2"
                          Whitespace@554..555 " "
                        Eq@555..556 "="
                        Whitespace@556..557 " "
                        EXPR_LET_VALUE@557..580
                          EXPR_CALL@557..580
                            EXPR_LIDENT@557..570
                              Lident@557..570 "triangle_type"
                            ARG_LIST@570..580
                              LParen@570..571 "("
                              ARG@571..574
                                EXPR_INT@571..572
                                  Int@571..572 "3"
                                Comma@572..573 ","
                                Whitespace@573..574 " "
                              ARG@574..577
                                EXPR_INT@574..575
                                  Int@574..575 "2"
                                Comma@575..576 ","
                                Whitespace@576..577 " "
                              ARG@577..578
                                EXPR_INT@577..578
                                  Int@577..578 "1"
                              RParen@578..579 ")"
                              Whitespace@579..580 " "
                        InKeyword@580..582 "in"
                        Whitespace@582..585 "\n  "
                        EXPR_LET_BODY@585..843
                          EXPR_LET@585..843
                            LetKeyword@585..588 "let"
                            Whitespace@588..589 " "
                            PATTERN_VARIABLE@589..596
                              Lident@589..595 "shape3"
                              Whitespace@595..596 " "
                            Eq@596..597 "="
                            Whitespace@597..598 " "
                            EXPR_LET_VALUE@598..621
                              EXPR_CALL@598..621
                                EXPR_LIDENT@598..611
                                  Lident@598..611 "triangle_type"
                                ARG_LIST@611..621
                                  LParen@611..612 "("
                                  ARG@612..615
                                    EXPR_INT@612..613
                                      Int@612..613 "2"
                                    Comma@613..614 ","
                                    Whitespace@614..615 " "
                                  ARG@615..618
                                    EXPR_INT@615..616
                                      Int@615..616 "3"
                                    Comma@616..617 ","
                                    Whitespace@617..618 " "
                                  ARG@618..619
                                    EXPR_INT@618..619
                                      Int@618..619 "2"
                                  RParen@619..620 ")"
                                  Whitespace@620..621 " "
                            InKeyword@621..623 "in"
                            Whitespace@623..626 "\n  "
                            EXPR_LET_BODY@626..843
                              EXPR_LET@626..843
                                LetKeyword@626..629 "let"
                                Whitespace@629..630 " "
                                PATTERN_WILDCARD@630..632
                                  WildcardKeyword@630..631 "_"
                                  Whitespace@631..632 " "
                                Eq@632..633 "="
                                Whitespace@633..634 " "
                                EXPR_LET_VALUE@634..656
                                  EXPR_CALL@634..656
                                    EXPR_LIDENT@634..648
                                      Lident@634..648 "string_println"
                                    ARG_LIST@648..656
                                      LParen@648..649 "("
                                      ARG@649..654
                                        EXPR_LIDENT@649..654
                                          Lident@649..654 "first"
                                      RParen@654..655 ")"
                                      Whitespace@655..656 " "
                                InKeyword@656..658 "in"
                                Whitespace@658..661 "\n  "
                                EXPR_LET_BODY@661..843
                                  EXPR_LET@661..843
                                    LetKeyword@661..664 "let"
                                    Whitespace@664..665 " "
                                    PATTERN_WILDCARD@665..667
                                      WildcardKeyword@665..666 "_"
                                      Whitespace@666..667 " "
                                    Eq@667..668 "="
                                    Whitespace@668..669 " "
                                    EXPR_LET_VALUE@669..692
                                      EXPR_CALL@669..692
                                        EXPR_LIDENT@669..683
                                          Lident@669..683 "string_println"
                                        ARG_LIST@683..692
                                          LParen@683..684 "("
                                          ARG@684..690
                                            EXPR_LIDENT@684..690
                                              Lident@684..690 "second"
                                          RParen@690..691 ")"
                                          Whitespace@691..692 " "
                                    InKeyword@692..694 "in"
                                    Whitespace@694..697 "\n  "
                                    EXPR_LET_BODY@697..843
                                      EXPR_LET@697..843
                                        LetKeyword@697..700 "let"
                                        Whitespace@700..701 " "
                                        PATTERN_WILDCARD@701..703
                                          WildcardKeyword@701..702 "_"
                                          Whitespace@702..703 " "
                                        Eq@703..704 "="
                                        Whitespace@704..705 " "
                                        EXPR_LET_VALUE@705..727
                                          EXPR_CALL@705..727
                                            EXPR_LIDENT@705..719
                                              Lident@705..719 "string_println"
                                            ARG_LIST@719..727
                                              LParen@719..720 "("
                                              ARG@720..725
                                                EXPR_LIDENT@720..725
                                                  Lident@720..725 "third"
                                              RParen@725..726 ")"
                                              Whitespace@726..727 " "
                                        InKeyword@727..729 "in"
                                        Whitespace@729..732 "\n  "
                                        EXPR_LET_BODY@732..843
                                          EXPR_LET@732..843
                                            LetKeyword@732..735 "let"
                                            Whitespace@735..736 " "
                                            PATTERN_WILDCARD@736..738
                                              WildcardKeyword@736..737 "_"
                                              Whitespace@737..738 " "
                                            Eq@738..739 "="
                                            Whitespace@739..740 " "
                                            EXPR_LET_VALUE@740..763
                                              EXPR_CALL@740..763
                                                EXPR_LIDENT@740..754
                                                  Lident@740..754 "string_println"
                                                ARG_LIST@754..763
                                                  LParen@754..755 "("
                                                  ARG@755..761
                                                    EXPR_LIDENT@755..761
                                                      Lident@755..761 "shape1"
                                                  RParen@761..762 ")"
                                                  Whitespace@762..763 " "
                                            InKeyword@763..765 "in"
                                            Whitespace@765..768 "\n  "
                                            EXPR_LET_BODY@768..843
                                              EXPR_LET@768..843
                                                LetKeyword@768..771 "let"
                                                Whitespace@771..772 " "
                                                PATTERN_WILDCARD@772..774
                                                  WildcardKeyword@772..773 "_"
                                                  Whitespace@773..774 " "
                                                Eq@774..775 "="
                                                Whitespace@775..776 " "
                                                EXPR_LET_VALUE@776..799
                                                  EXPR_CALL@776..799
                                                    EXPR_LIDENT@776..790
                                                      Lident@776..790 "string_println"
                                                    ARG_LIST@790..799
                                                      LParen@790..791 "("
                                                      ARG@791..797
                                                        EXPR_LIDENT@791..797
                                                          Lident@791..797 "shape2"
                                                      RParen@797..798 ")"
                                                      Whitespace@798..799 " "
                                                InKeyword@799..801 "in"
                                                Whitespace@801..804 "\n  "
                                                EXPR_LET_BODY@804..843
                                                  EXPR_LET@804..843
                                                    LetKeyword@804..807 "let"
                                                    Whitespace@807..808 " "
                                                    PATTERN_WILDCARD@808..810
                                                      WildcardKeyword@808..809 "_"
                                                      Whitespace@809..810 " "
                                                    Eq@810..811 "="
                                                    Whitespace@811..812 " "
                                                    EXPR_LET_VALUE@812..835
                                                      EXPR_CALL@812..835
                                                        EXPR_LIDENT@812..826
                                                          Lident@812..826 "string_println"
                                                        ARG_LIST@826..835
                                                          LParen@826..827 "("
                                                          ARG@827..833
                                                            EXPR_LIDENT@827..833
                                                              Lident@827..833 "shape3"
                                                          RParen@833..834 ")"
                                                          Whitespace@834..835 " "
                                                    InKeyword@835..837 "in"
                                                    Whitespace@837..840 "\n  "
                                                    EXPR_LET_BODY@840..843
                                                      EXPR_UNIT@840..843
                                                        LParen@840..841 "("
                                                        RParen@841..842 ")"
                                                        Whitespace@842..843 "\n"
      RBrace@843..844 "}"
      Whitespace@844..845 "\n"