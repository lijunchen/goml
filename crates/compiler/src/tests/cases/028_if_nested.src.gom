package main

import (
    "fmt"
)

func int_neg(x int) int {
    return -x
}

func int_less(x int, y int) bool {
    return x < y
}

func string_println(s string) struct{} {
    fmt.Println(s)
    return struct{}{}
}

func classify(x__0 int) string {
    var ret12 string
    var t6 bool = int_less(x__0, 0)
    if t6 {
        ret12 = "negative"
    } else {
        var t7 bool = int_less(0, x__0)
        if t7 {
            ret12 = "positive"
        } else {
            ret12 = "zero"
        }
    }
    return ret12
}

func triangle_type(a__1 int, b__2 int, c__3 int) string {
    var ret13 string
    var t8 bool = int_less(a__1, b__2)
    if t8 {
        var t9 bool = int_less(b__2, c__3)
        if t9 {
            ret13 = "ascending"
        } else {
            ret13 = "peak"
        }
    } else {
        var t10 bool = int_less(a__1, c__3)
        if t10 {
            ret13 = "valley"
        } else {
            ret13 = "flat"
        }
    }
    return ret13
}

func main0() struct{} {
    var ret14 struct{}
    var t11 int = int_neg(42)
    var first__4 string = classify(t11)
    var second__5 string = classify(0)
    var third__6 string = classify(17)
    var shape1__7 string = triangle_type(1, 2, 3)
    var shape2__8 string = triangle_type(3, 2, 1)
    var shape3__9 string = triangle_type(2, 3, 2)
    string_println(first__4)
    string_println(second__5)
    string_println(third__6)
    string_println(shape1__7)
    string_println(shape2__8)
    string_println(shape3__9)
    ret14 = struct{}{}
    return ret14
}

func main() {
    main0()
}
