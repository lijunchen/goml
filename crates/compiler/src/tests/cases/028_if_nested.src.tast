fn classify(x/0: int) -> string {
  if int_less((x/0 : int), 0) {
      "negative"
  } else {
      if int_less(0, (x/0 : int)) {
          "positive"
      } else {
          "zero"
      }
  }
}

fn triangle_type(a/1: int, b/2: int, c/3: int) -> string {
  if int_less((a/1 : int), (b/2 : int)) {
      if int_less((b/2 : int), (c/3 : int)) {
          "ascending"
      } else {
          "peak"
      }
  } else {
      if int_less((a/1 : int), (c/3 : int)) {
          "valley"
      } else {
          "flat"
      }
  }
}

fn main() -> unit {
  let first/4: string = classify(-42) in
  let second/5: string = classify(0) in
  let third/6: string = classify(17) in
  let shape1/7: string = triangle_type(1, 2, 3) in
  let shape2/8: string = triangle_type(3, 2, 1) in
  let shape3/9: string = triangle_type(2, 3, 2) in
  let _ : unit = string_println((first/4 : string)) in
  let _ : unit = string_println((second/5 : string)) in
  let _ : unit = string_println((third/6 : string)) in
  let _ : unit = string_println((shape1/7 : string)) in
  let _ : unit = string_println((shape2/8 : string)) in
  let _ : unit = string_println((shape3/9 : string)) in
  ()
}