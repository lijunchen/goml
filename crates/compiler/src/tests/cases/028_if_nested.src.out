negative
zero
positive
ascending
flat
peak
